src/main/java/org/apache/cassandra/sidecar/config/yaml/MetricsFilteringConfigurationImpl.java

```
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.config.yaml;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.cassandra.sidecar.config.MetricsFilteringConfiguration;
+
+/**
+ * Holds configuration needed for filtering out metrics
+ */
+public class MetricsFilteringConfigurationImpl implements MetricsFilteringConfiguration
+{
+    public static final String DEFAULT_TYPE = "regex";
+    public static final String DEFAULT_VALUE = ".*"; // include all metrics
+    @JsonProperty("type")
+    private final String type;
+    @JsonProperty("value")
+    private final String value;
+
+    public MetricsFilteringConfigurationImpl()
+    {
+        this(DEFAULT_TYPE, DEFAULT_VALUE);
+    }
+
+    public MetricsFilteringConfigurationImpl(String type, String value)
+    {
+        this.type = type;
+        this.value = value;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String type()
+    {
+        return type;
```

type is not validated. We should validate the allowed strings.

---

You want to fail? We validate while parsing and then only create. If they dont match expected, we function as if that filter was not valid

---

I think it is a reasonable expectation to fail the service startup if misconfigured. As opposed to ignoring the error and have an operator spend a lot of time trying to figure out why something they think should be working is not really working. Take a look at this for example: [testInvalidClientAuth](https://github.com/apache/cassandra-sidecar/blob/trunk/src/test/java/org/apache/cassandra/sidecar/config/SidecarConfigurationTest.java#L176-L184)

---

Got it will add the check to configImpl